apps/ejabberd/src/node_push.erl

```diff
@@ -0,0 +1,226 @@
+%%%-------------------------------------------------------------------
+%%% @author Rafal Slota
+%%% @copyright (C) 2017 Erlang Solutions Ltd.
+%%% This software is released under the Apache License, Version 2.0
+%%% cited in 'LICENSE.txt'.
+%%% @end
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% @todo: Write me!
+%%% @end
+%%%-------------------------------------------------------------------
+-module(node_push).
+-author('rafal.slota@erlang-solutions.com').
+-behaviour(gen_pubsub_node).
+
+-include_lib("ejabberd/include/ejabberd.hrl").
+-include_lib("ejabberd/include/jlib.hrl").
+-include_lib("ejabberd/include/pubsub.hrl").
+
+-export([init/3, terminate/2, options/0, features/0,
+         create_node_permission/6, create_node/2, delete_node/1,
+         purge_node/2, subscribe_node/8, unsubscribe_node/4,
+         publish_item/8, delete_item/4, remove_extra_items/3,
+         get_entity_affiliations/2, get_node_affiliations/1,
+         get_affiliation/2, set_affiliation/3,
+         get_entity_subscriptions/2, get_node_subscriptions/1,
+         get_subscriptions/2, set_subscriptions/4,
+         get_pending_nodes/2, get_states/1, get_state/2,
+         set_state/1, get_items/7, get_items/3, get_item/7,
+         get_item/2, set_item/1, get_item_name/3, node_to_path/1,
+         path_to_node/1]).
+
+init(Host, ServerHost, Opts) ->
+    node_flat:init(Host, ServerHost, Opts),
```

Comment:
How about we try [inaka/mixer](https://github.com/inaka/mixer) to reduce number of LOC in this module?